beacon-chain/p2p/iterator.go

```diff
@@ -0,0 +1,32 @@
+package p2p
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+)
+
+// filterNodes wraps an iterator such that Next only returns nodes for which
+// the 'check' function returns true. This custom implementation also
+// checks for context deadlines so that t
```

Comment:
godoc incomplete. It would be nice to have an explanation for why we are writing a custom iterator in Prysm